File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc(a, b):
    return [i for i in range(a, b + 1)] 

def applicatorFunc(inpFunc, s, a, b):
    if s == 's':
        return sum(inpFunc(a, b))  # Return the sum
    else:
        return sum(inpFunc(a, b)) / len(inpFunc(a, b))  # Return the average

a = 1
b =5


result = applicatorFunc(listFunc, 'a', a, b)
print("Result =", result)

File: lab2-functionalprimer/haskell/func_as_arg.hs
-- Define a function to generate a list of integers from a to b
inpFunc a b = [a..b]

applicatorFunc inpFunc s
    | s == 's'  = sum inpFunc             
    | otherwise = (sum inpFunc) / fromIntegral (length inpFunc)

main = do
  
    let a = 1
    let b = 5
    let result = applicatorFunc (inpFunc a b) 's' -- Call applicatorFunc with inpFunc and 'a' as args
    putStrLn("Result = " ++ show(result))